sockeye/loss.py

```diff
@@ -35,51 +35,84 @@ class LossConfig(config.Config):
     :param vocab_size: Target vocab size.
     :param normalization_type: How to normalize the loss.
     :param label_smoothing: Optional smoothing constant for label smoothing.
+    :param link: Link function.
+    :param weight: Loss weight.
     """
 
     def __init__(self,
                  name: str,
-                 vocab_size: int,
-                 normalization_type: str,
-                 label_smoothing: float = 0.0) -> None:
+                 vocab_size: Optional[int] = None,
+                 normalization_type: Optional[str] = None,
+                 label_smoothing: float = 0.0,
+                 link: Optional[str] = None,
```

Comment:
these two new parameters should be prefixed with something like 'length_task_' to make clear that they are not for the main loss.
In the future we need to rewrite this anyway.

Reply:
I am also not sure we should call this 'link'. Why not just loss and name the options poisson and mse?

Reply:
Prepended the var names. 

Technically, `link` is more precise because the loss is the same in both cases -- MLE, it just looks differently because of different distributions.

Reply:
good point, let's leave 'link'